diff --git a/src/Authentication/TokenAuthUser.php b/src/Authentication/TokenAuthUser.php
index 0e2c34d..9e7ab9c 100644
--- a/src/Authentication/TokenAuthUser.php
+++ b/src/Authentication/TokenAuthUser.php
@@ -911,4 +911,18 @@ class TokenAuthUser implements TokenAuthUserInterface {
     return $this->subject->isSyncing();
   }
 
+   /**
+   * {@inheritdoc}
+   */
+  public function setLanguageAware($flag = TRUE) {
+    return $this->subject->setLanguageAware($flag = TRUE);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isLanguageAware() {
+    return $this->subject->isLanguageAware();
+  }
+
 }
